pkg/engine/systemroleassignments.go

```diff
@@ -88,13 +90,24 @@ func createKubernetesMasterRoleAssignmentForAgentPools(masterProfile *api.Master
 		dependenciesToMasterVms[masterIdx] = fmt.Sprintf("[concat(variables('masterVMNamePrefix'), %d)]", masterIdx)
 	}
 
-	var roleAssignmentsForAllAgentPools = make([]DeploymentWithResourceGroupARM, len(agentPoolProfiles))
+	roleAssignmentsForAllAgentPools := []DeploymentWithResourceGroupARM{}
 
 	// The following is based on:
 	// * https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/role-based-access-control/role-assignments-template.md#create-a-role-assignment-at-a-resource-scope
 	// * https://github.com/Azure/azure-quickstart-templates/blob/master/201-rbac-builtinrole-multipleVMs/azuredeploy.json#L79
-	for agentPoolIdx, agentPool := range agentPoolProfiles {
+	found := []string{}
+	for _, agentPool := range agentPoolProfiles {
 		var roleAssignments = make([]interface{}, masterProfile.Count)
+		subnetElements := strings.Split(agentPool.VnetSubnetID, "/")
+		if len(subnetElements) < 9 {
+			continue
+		}
+		vnet := strings.Join(subnetElements[:9], "/")
+		if helpers.IsStringInStringSlice(vnet, found) {
```

Comment:
instead of having to loop through the string slice everytime you could use a `map[string]struct{}` which is more performant and doesn't require the addionnal helper func, ie. https://www.davidkaya.com/sets-in-golang/

Reply:
Thanks for the optimization! Done